api/v1alpha1/clusterconfig_types.go

```diff
@@ -186,6 +191,40 @@ func (ExtraAPIServerCertSANs) VariableSchema() clusterv1.VariableSchema {
 	}
 }
 
+// ImageRegistryCredentials required for providing credentials for an image registry URL.
+type ImageRegistryCredentials struct {
+	// Registry URL.
+	URL string `json:"url"`
+
+	// Name of the Secret containing the registry credentials.
+	// The Secret should have keys 'username' and 'password'.
+	// This credentials Secret is not required for some registries, e.g. ECR.
+	// +optional
+	Secret string `json:"secret,omitempty"`
```

Comment:
Looking for suggestions on what this should look like 🙏

Reply:
I think making this a `secretRef` (like an `ObjectReference`) just with `name` field for now would make more sense. If we make that it's own struct then we'll be able to reuse that as well.